app/controllers/volunteers_controller.rb

```diff
@@ -67,11 +66,41 @@ def remove
     # no longer available?
   end
 
-private
-    def volunteer_params
-      params.require(:volunteer).permit(:picture,:first_name,:last_name,:email,
-      :mobile_phone, :birthdate, :gender, :city, :state, :employer, :position,
-      :availability, :skill, :volunteer_notes)
-    end
+  def add_repairs
+    @repair = Repair.all
+    if current_user.nil?
+      redirect_to new_volunteer_path
+    else
+      @volunteer = current_user.volunteer
+  end
+  end
+
+  def add_repair_to_volunteer
+    # @repair = Repair.find(params[:id])
+    volunteer_repair = VolunteerRepair.new
+    if current_user.nil?
```

Comment:
Again, this line should say: if current_user.volunteer.nil?  current_user will not be nil, but the user may have no volunteer attached.